testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java

```diff
@@ -55,20 +55,20 @@ public void testColumnMappingModeNone()
 
         onDelta().executeQuery("" +
                 "CREATE TABLE default." + tableName +
-                " (a_number INT)" +
+                " (a_number INT, nested STRUCT<field1: STRING, field2: STRING>)" +
```

Comment:
This looks like the only product test modified.
We should have tests for tables that underwent insert/update/delete/merge with onDelta + dereference pushdown (for all column mapping modes)

can be follow-up, but this is probably even more important than those future-proofing test cases being added in integrstion tests

Reply:
I will remove this change and create a follow-up.

Reply:
create a follow-up with ... ?

are we keeping and expanding product test coverage within this PR?

Reply:
> follow-up with ... ?

tables that underwent insert/update/delete/merge with onDelta + dereference pushdown.

I think I will add a separate test for the above use case to the current PR only.

Reply:
Existing test case:

`testColumnMappingModeNone` -> covers `none` with tables that underwent only insert
`testColumnMappingMode` -> covers `id`, `name` with tables that underwent only insert

New test case:
`testProjectionPushdownDmlWithColumnMappingMode` -> covers `id`, `name` with tables that underwent insert/update/delete/merge with onDelta

Reply:
> We should have tests for tables that underwent insert/update/delete/merge with onDelta + dereference pushdown (for all column mapping modes)

@findepi  what is the rationale for this request in the context of this _read_ oriented feature ? Is this more about being future-proof?